src/adapter/service-worker/index.ts

```diff
@@ -2,4 +2,29 @@
  * Service Worker Adapter for Hono.
  * @module
  */
-export { handle } from './handler'
+import type { Hono } from '../../hono'
+import { handle } from './handler'
+
+/**
+ * Registers a Hono app to handle fetch events in a service worker.
+ * This sets up `addEventListener('fetch', handle(app))` for the provided app.
+ *
+ * @param app - The Hono application instance
+ * @example
+ * ```ts
+ * import { Hono } from 'hono'
+ * import { fire } from 'hono/service-worker'
+ *
+ * const app = new Hono()
+ *
+ * app.get('/', (c) => c.text('Hi'))
+ *
+ * fire(app)
+ * ```
+ */
+const fire = (app: Hono): void => {
+  // @ts-expect-error addEventListener is not typed well
+  addEventListener('fetch', handle(app))
+}
```

Comment:
For compatibility with `base fetch`:

```suggestion
const fire = (app: Hono): void => {
  // @ts-expect-error addEventListener is not typed well
  addEventListener('fetch', handle(app. {fetch: undefined}))
}
```

For usage in either web worker or web server, but a potentially unexpected default for web server:

```suggestion
const fire = (app: Hono, opts: HandleOpts): void => {
  // @ts-expect-error addEventListener is not typed well
  addEventListener('fetch', handle(app, opts))
}
```

Reply:
another idea could be to have separate methods like:
- `fire()`: which is compatible with `base fire` and passes `handle(app, {fetch: undefined})`
- and `fireWithFallback(opts)`: which can be configure for either case and uses the service worker default fallback fetch behavior